src/__tests__/merge.test.ts

```diff
@@ -43,49 +43,99 @@ describe('mergeResponses()', () => {
       {
         code: '200',
         headers: [
-          { name: '200h1', style: HttpParamStyles.Simple },
-          { name: '200h2', style: HttpParamStyles.Simple },
-          { name: '200h3', style: HttpParamStyles.Simple },
+          { name: '200h1', style: HttpParamStyles.Simple, required: false },
+          { name: '200h2', style: HttpParamStyles.Simple, required: true },
+          { name: '200h3', style: HttpParamStyles.Simple, required: false },
         ],
         contents: [],
       },
       {
         code: '400',
         headers: [
-          { name: '400h1', style: HttpParamStyles.Simple },
-          { name: '400h2', style: HttpParamStyles.Simple },
+          { name: '400h1', style: HttpParamStyles.Simple, required: true },
+          { name: '400h2', style: HttpParamStyles.Simple, required: true },
         ],
       },
       {
         code: '500',
         headers: [
-          { name: '500h1', style: HttpParamStyles.Simple },
-          { name: '500h2', style: HttpParamStyles.Simple },
+          { name: '500h1', style: HttpParamStyles.Simple, required: true },
```

Comment:
This test merges four responses. Two 200's, one 400 and one 500.
Because of that, those 400 and 500 stays untouched and those 200's are merged together with proper `required` flag (depending on the presence of headers in each of them).

Reply:
Oh now I understand. This is the `toEqual` part 👍 
This is ok!